core/src/main/scala/variant/Variant.scala

```diff
@@ -153,11 +153,14 @@ abstract class Variant private[variant] (
   def opponentHasInsufficientMaterial(position: Position): Boolean =
     InsufficientMatingMaterial(position, !position.color)
 
+  def playerHasInsufficientMaterial(position: Position): Boolean =
+    throw Exception("`playerHasInsufficientMaterial` is not implemented for this variant")
```

Comment:
We usually don't throw exceptions. If this function is only applicable to Antichess, maybe we should have this as public function in `Antichess`.

But tbh, I'm not sure. Do you think the issue we had in https://github.com/lichess-org/lila/pull/17114 only happen in Antichess? I haven't verified but I think it could happen to other variants as well.

Reply:
In `Horde.scala`, there's this code:

![image](https://github.com/user-attachments/assets/ce1fa0a2-63ca-4202-956f-9dee8102d737)

The comment mentions imminent fortresses aren't considered, but if they were then turn would matter.